client/allocrunner/network_hook.go

```diff
@@ -95,8 +113,25 @@ func (h *networkHook) Prerun() error {
 		return nil
 	}
 
-	spec, created, err := h.manager.CreateNetwork(h.alloc.ID)
+	// Perform our networks block interpolation.
+	interpolatedNetworks := taskenv.InterpolateNetworks(h.taskEnvBuilder.Build(), tg.Networks)
```

Comment:
Since you only read from the environment, pass a `taskenv.TaskEnv` into the hook instead of a builder. Not a big deal, but environment mutations can be tricky to trace. The 2 different types - Builder for mutating, TaskEnv for reading - help make it more obvious who mutates and who just reads the environment.